gpii/node_modules/dbOperation/test/DbDataStoreTestsUtils.js

```diff
@@ -0,0 +1,326 @@
+/**
+GPII DB Data Store Tests
+
+Copyright 2016-2017 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    jqUnit = fluid.require("node-jqunit", require, "jqUnit");
+
+require("./js/DataStoreTestsUtils.js");
+
+fluid.defaults("gpii.tests.dbDataStore.environment", {
+    gradeNames: ["gpii.test.pouch.environment"],
+    port: 1234,
+    pouchConfig: {
+        databases: {
+            gpii: {
+                data: [
+                    "%dbOperation/test/data/clientCredentials.json",
+                    "%dbOperation/test/data/gpiiAppInstallationAuthorizations.json",
+                    "%dbOperation/test/data/gpiiAppInstallationClients.json",
+                    "%dbOperation/test/data/gpiiKeys.json",
+                    "%dbOperation/test/data/prefsSafes.json",
+                    "%gpii-universal/testData/dbData/views.json"
+                ]
+            }
+        }
+    },
+    components: {
+        testCaseHolder: {
+            type: "gpii.tests.dbDataStore.baseTestCaseHolder"
+        }
+    },
+    distributeOptions: {
+        source: "{that}.options.rawModules",
+        target: "{that > testCaseHolder}.options.rawModules"
+    },
+    mergePolicy: {
+        rawModules: "noexpand"
+    }
+});
+
+fluid.defaults("gpii.tests.dbDataStore.baseTestCaseHolder", {
+    gradeNames: ["gpii.test.express.caseHolder"],
+    events: {
+        onResponse: null,
+        onError: null
+    },
+    components: {
+        dbDataStore: {
+            type: "gpii.dbOperation.dbDataStore",
+            options: {
+                dataSourceConfig: {
+                    baseUrl: "http://localhost",
+                    port: "{gpii.tests.dbDataStore.environment}.options.port",
+                    dbName: "gpii"
+                }
+            }
+        }
+    },
+    sequenceEnd: [{
+        func: "{gpii.tests.dbDataStore.environment}.events.onCleanup.fire"
+    }, {
+        event:    "{gpii.tests.dbDataStore.environment}.events.onCleanupComplete",
+        listener: "fluid.log",
+        args:     ["Database cleanup complete"]
+    }]
+});
+
+gpii.tests.dbDataStore.saveAndInvokeFetch = function (fetchDataSource, id, that) {
+    gpii.tests.dbDataStore.lastSavedId = id;
```

Comment:
Could we store this `id` in a more localised place? Such as on the testCase? Rather than in a global.

Using a global will cause issues in the future if we have concurrent tests that share this state.